dev/WindowsAppRuntime_MSIXInstallFromPath/main.cpp

```diff
@@ -0,0 +1,303 @@
+            L"           => Microsoft.WindowsAppRuntime.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.DDLM.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Main.*.msix\n"
+            L"           => Microsoft.WindowsAppRuntime.Singleton.*.msix\n");
+}
+
+HRESULT JustDoIt(PCWSTR path) noexcept try
+{
+    wprintf(L"path: %s\n", path);
+
+    // Install packages
+    auto fpath{ std::filesystem::path(path) };
+    auto inventory{ fpath / L"MSIX.inventory" };
+    std::string lineUtf8;
+    std::ifstream f{ inventory };
+    while (getline(f, lineUtf8))
+    {
+        // Skip blank lines
+        if (lineUtf8.empty())
+        {
+            continue;
+        }
+
+        // Skip comments
+        if (lineUtf8[0] == '#')
+        {
+            continue;
+        }
+
+        // Parse the line
+        auto line{ Microsoft::Utf8::ToUtf16<std::wstring>(lineUtf8.c_str()) };
+        auto offset{ line.find(L'=') };
+        THROW_HR_IF_MSG(E_UNEXPECTED, offset == std::wstring::npos, "line:%s", lineUtf8.c_str());
+        auto filename{ line.substr(0, offset) };
+        THROW_HR_IF_MSG(E_UNEXPECTED, filename.empty(), "line:%s", lineUtf8.c_str());
+        auto packageFullName{ std::wstring(line.c_str() + offset + 1) };
+        THROW_HR_IF_MSG(E_UNEXPECTED, packageFullName.empty(), "line:%s", lineUtf8.c_str());
+
+        AddPackageIfNecessary(path, filename, packageFullName);
+    }
+
+    return 0;
+}
+CATCH_RETURN();
+
+void AddPackageIfNecessary(PCWSTR path, const std::wstring& filename, const std::wstring& packageFullName)
+{
+    wprintf(L"Path: %s\n", path);
+    wprintf(L"Filename: %s\n", filename.c_str());
+    wprintf(L"PackageFullName: %s\n", packageFullName.c_str());
+
+    if (!NeedToRegisterPackage(packageFullName))
+    {
+        return;
+    }
+
+    auto hr{ AddPackage(path, filename) };
+    if (FAILED(hr))
+    {
+        wprintf(L"AddPackage(): 0x%X Path:%ls Filename:%ls PackageFullName:%ls", hr, path, filename.c_str(), packageFullName.c_str());
+        THROW_HR_MSG(hr, "Path:%ls Filename:%ls PackageFullName:%ls", path, filename.c_str(), packageFullName.c_str());
+    }
+}
+
+bool NeedToRegisterPackage(const std::wstring& packageFullName)
+{
+    return NeedToRegisterPackage(packageFullName.c_str());
+}
+
+bool NeedToRegisterPackage(PCWSTR packageFullName)
+{
+    auto packageId{ PackageId::FromPackageFullName(packageFullName) };
+
+    // Is there an equal-or-better package already registered?
+    wprintf(L"Family: %s\n", packageId.PackageFamilyName());
+    auto packageFullNames{ AppModel::Package::FindByFamily(packageId.PackageFamilyName()) };
+    if (packageFullNames.empty())
+    {
+        wprintf(L"1 None found\n");
+        return true;
+    }
+    for (auto registeredPackageFullName : packageFullNames)
+    {
+        auto registeredPackage{ PackageId::FromPackageFullName(registeredPackageFullName) };
+        wprintf(L"C %ls %ls %d %d %I64X %I64X\n",
```

Comment:
Debugging printfer will go away in the follow up PR after VS invoking this is tested